src/dune_rules/melange/melange.mli

```diff
@@ -5,6 +5,8 @@ module Module_system : sig
     | Es6
     | CommonJs
 
+  val default : t * string
```

Comment:
Could you introduce a type alias` Filename.Extension.t` and use it here along with some other places in this PR? Otherwise, the code is a bit hard to read.